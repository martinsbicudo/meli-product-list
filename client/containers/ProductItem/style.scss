.ProductItem {
  width: 100%;
  flex: 1;
  display: flex;
  box-sizing: border-box;

  &Figure {
    margin: -2rem;
    margin-right: 2rem;
    object-fit: cover;
    display: flex;
    align-items: center;
  }

  &Price {
    font-size: 1.6rem;
    margin-bottom: 1rem;
    font-weight: bolder;

    span {
      color: map-get($colors, "meli-dodger-blue");
      margin-right: .2rem;
    }
  }

  &Title {
    font-size: 1.4rem;
  }
}
