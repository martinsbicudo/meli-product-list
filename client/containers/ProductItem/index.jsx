import React from 'react';
import { object } from 'prop-types';
import CardBoard from '@Components/CardBoard';
import style from './style.scss';

const ProductItem = ({
  product,
}) => (
  <CardBoard className={style.ProductItem}>
    <figure className={style.ProductItemFigure}>
      <img
        src={product.thumbnail}
        alt={product.title}
        title={product.title}
      />
    </figure>
    <div>
      <p className={style.ProductItemPrice}>
        <span>$</span>
        {product.price}
      </p>
      <h2 className={style.ProductItemTitle}>
        {product.title}
      </h2>
    </div>
  </CardBoard>
);

ProductItem.propTypes = {
  product: object.isRequired,
};

export default ProductItem;
