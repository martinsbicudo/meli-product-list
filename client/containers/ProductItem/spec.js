import ProductItem from '.';

describe('Containers - <ProductItem />', () => {
  const testMessageTitle = 'TEST_TITLE';
  const testMessagePrice = '300';
  const testMessageImage = 'TEST_IMAGE';
  const createWrapper = () => render(
    <ProductItem
      product={{
        title: testMessageTitle,
        price: testMessagePrice,
        thumbnail: testMessageImage,
      }}
    />,
  );

  test('Should finds Title', () => {
    const { getByText } = createWrapper();

    expect(getByText(testMessageTitle)).toBeInTheDocument();
  });

  test('Should finds Image', () => {
    const { getByText } = createWrapper();

    expect(getByText(testMessagePrice)).toBeInTheDocument();
  });

  test('Should finds Description', () => {
    const { getByAltText } = createWrapper();

    expect(getByAltText(testMessageTitle).src).toBe(`http://localhost/${testMessageImage}`);
  });

  test('Wrapper Snapshot', () => {
    const { asFragment } = createWrapper();

    expect(asFragment()).toMatchSnapshot();
  });
});
