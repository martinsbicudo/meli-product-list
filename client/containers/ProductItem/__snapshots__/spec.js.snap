// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Containers - <ProductItem /> Wrapper Snapshot 1`] = `
<DocumentFragment>
  <div
    class="CardBoard ProductItem"
  >
    <figure
      class="ProductItemFigure"
    >
      <img
        alt="TEST_TITLE"
        src="TEST_IMAGE"
        title="TEST_TITLE"
      />
    </figure>
    <div>
      <p
        class="ProductItemPrice"
      >
        <span>
          $
        </span>
        300
      </p>
      <h2
        class="ProductItemTitle"
      >
        TEST_TITLE
      </h2>
    </div>
  </div>
</DocumentFragment>
`;
